app/controllers/journeys/capitalallowances/balancingCharge/BalancingChargeAmountController.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.journeys.capitalallowances.balancingCharge
+
+import cats.implicits.catsSyntaxOptionId
+import controllers.actions.{DataRequiredAction, DataRetrievalAction, IdentifierAction}
+import forms.capitalallowances.balancingCharge.BalancingChargeAmountFormProvider
+import models.Mode
+import models.common.{BusinessId, TaxYear}
+import navigation.CapitalAllowancesNavigator
+import pages.capitalallowances.balancingCharge.BalancingChargeAmountPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.SelfEmploymentService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.Logging
+import views.html.journeys.capitalallowances.balancingCharge.BalancingChargeAmountView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BalancingChargeAmountController @Inject() (override val messagesApi: MessagesApi,
+                                                 navigator: CapitalAllowancesNavigator,
+                                                 identify: IdentifierAction,
+                                                 getData: DataRetrievalAction,
+                                                 requireData: DataRequiredAction,
+                                                 service: SelfEmploymentService,
+                                                 formProvider: BalancingChargeAmountFormProvider,
```

Comment:
Can the BigDecimalFormProvider be used here instead?

Reply:
Probably - I just used the same pattern as previous Controllers which have a custom Form